include/dlaf/util_matrix.h

```diff
@@ -21,149 +21,111 @@ constexpr double M_PI = 3.141592;
-    else if (op == blas::Op::Trans || op == blas::Op::ConjTrans) {
-      if (mat_a.nrTiles().rows() != mat_b.nrTiles().rows()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not left multipliable (cols of matrix A not equal to rows of matrix B).");
-      }
-      if (mat_a.size().rows() != mat_b.size().rows()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not left multipliable (size of matrix A not equal to that of matrix B).");
-      }
-      if (mat_a.blockSize().rows() != mat_b.blockSize().rows()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not left multipliable (blocksize of matrix A not equal to that of matrix B).");
-      }
-    }
-  }
-  else if (side == blas::Side::Right) {
-    if (op == blas::Op::NoTrans) {
-      if (mat_a.nrTiles().rows() != mat_b.nrTiles().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (rows of matrix A not equal to cols of matrix B).");
-      }
-      if (mat_a.size().rows() != mat_b.size().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (size of matrix A not equal to that of matrix B).");
-      }
-      if (mat_a.blockSize().rows() != mat_b.blockSize().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (blocksize of matrix A not equal to that of matrix B).");
-      }
-    }
-    else if (op == blas::Op::Trans || op == blas::Op::ConjTrans) {
-      if (mat_a.nrTiles().cols() != mat_b.nrTiles().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (rows of matrix A not equal to cols of matrix B).");
-      }
-      if (mat_a.size().cols() != mat_b.size().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (size of matrix A not equal to that of matrix B).");
-      }
-      if (mat_a.blockSize().cols() != mat_b.blockSize().cols()) {
-        throw std::invalid_argument(
-            function + ": " + "The matrices " + mat_a_name + " and " + mat_b_name +
-            " are not right multipliable (blocksize of matrix A not equal to that of matrix B).");
-      }
-    }
-  }
-}
-
-}
-
-namespace matrix {
-namespace util {
-namespace internal {
+#define DLAF_PRECONDITION_MULTIPLIABLE_MATRICES(a, b, opA, opB) \
```

Comment:
I would add matrix c as well to have a full check that a*b can be performed and stored to c.